### PR TITLE
Add Docs for Dev Environments

### PR DESCRIPTION
**Pull request by usha-mandya**
Signed-off-by: Usha Mandya <usha.mandya@docker.com>

Docs for Dev Env Preview

### CODE REVIEW

**Review by glours — DISMISSED** [review]
LGTM

File: desktop/dev-environments.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+![Single container Dev environment](/images/dev-env-container.png){:width="700px"}
+
+In the above example, the names `amazing_mclaren` and `stoic_carver` are randomly generated. You'll most likely see different names when you create your Dev Environment.
+
+Hover over the container and click **Open in VS Code** to start working in VS Code as usual. You can also open a terminal in VS Code, and use Git to push or pull code to your repository, or switch between branches and work as you would normally.
+
+You can launch the application by running the command `make run` in your VS Code terminal. This opens an http server on port 8080. Open [http://localhost:8080](http://localhost:8080) in your browser to see the running application.
+
+![Single container Dev environment](/images/dev-env-localhost.png){:width="700px"}
+
```
Comment (gtardif) [inline desktop/dev-environments.md:54]: Maybe emphasis a bit more the magical parts after this first step ?
* the repo was cloned in a container using your git credentials, and does not collide with any of you local files on your host
* you don't need to install any tools, like go tools, all this is setup automatically, and also does not collide with any of you local tooling
* You can run the app server (inside a container) from the VSCode terminal and it is accessible from the host automatically

Comment (usha-mandya) [reply desktop/dev-environments.md:54]: @gtardif PTAL

**Review by glours — DISMISSED** [review]

**Review by gtardif — APPROVED** [review]

File: desktop/dev-environments.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+CMD ["/server"]
+```
+
+The `development`target uses a `golang:1.16-alpine` image with all dependencies you need for development. You can start your project directly from VS Code and interact with the others applications or services such as the database or the frontend.
+
+In our example, the Docker Compose files are the same. However, they could be different and the services defined in the main Compose file may use other targets to build or directly reference other images.
+
+## Specify a base image
+
+In this preview, Dev Environments support a simple YAML file which allows you to specify the base image that you would like to use as part of your Dev Environment. You must include this as part of the `.docker` folder and then add it as a `config.json` file. For example, to use the Jekyll base image, add:
```
Comment (rumpl) [inline desktop/dev-environments.md:176]:
```suggestion
In this preview, Dev Environments support a simple JSON file which allows you to specify the base image that you would like to use as part of your Dev Environment. You must include this as part of the `.docker` folder and then add it as a `config.json` file. For example, to use the Jekyll base image, add:
```

**Review by rumpl — APPROVED** [review]
One small nit but otherwise LGTM 👍

**Review by rumpl — DISMISSED** [review]

**Review by crydotsnake — APPROVED** [review]

**Review by glours — APPROVED** [review]

**Review by nebuk89 — APPROVED** [review]
LGTM - ship it :D

**Pull request merged**